src/migrations.js

```diff
@@ -24,7 +21,7 @@ require.scopes.migrations = (function() {
 var exports = {};
 exports.Migrations= {
   changePrivacySettings: function() {
-    if (!chrome.extension.inIncognitoContext) {
+    if (!chrome.extension.inIncognitoContext && chrome.privacy ) {
```

Comment:
Actually, the method above was working. There were other issues with the popup display which made me think otherwise.

My proposed solution: Check for incognito mode in a content script on each tab load, add this to `pb.tabData[tab.id].inIncognito`, then we'll refer to this `inIncognito` property to check for ingcognito mode.